app/policies/article_policy.rb

```diff
@@ -12,6 +12,23 @@ def self.limit_post_creation_to_admins?
     FeatureFlag.enabled?(:limit_post_creation_to_admins)
   end
 
+  # @param query [Symbol] the name of one of the ArticlePolicy action predicates (e.g. :create?,
+  #        :new?) though as a convenience, we will also accept :new, and :create.
+  # @return [TrueClass] if this query should default to hidden
+  # @return [FalseClass] if this query should not be hidden in the UI.
+  #
+  # @note The symmetry in case logic as in .scope_users_authorized_to_action; a future path might be
+  # to look at how controller actions do the 'respond_to do' blocks.  But that's already a bit
+  # further afield than this need be.
```

Comment:
@jeremyf Looking at this comment, I find that this comment talks about a future that "may" never happen which can deem the comment stale at some point. The comment also doesn't serve as an executable, so maybe this is a better fit for a "future improvement" project board instead of living in the codebase?

I'd also like to chat about the our broader usage of comments. I've found the addition of comments that are explaining "why" (if it's not obvious) to be very useful. However, with the increased level of comments in our codebase recently, I've also found myself either distracted by some the comments or skipping over them, thus negating their purpose. In addition, having too many comments/notes result in an increased overhead where we need to maintain the methods and the comments and perhaps we can come up with other ways to convey this knowledge.

Maybe we can work together to establish a common language on what we think makes for best practice comments?

Reply:
Thank you @ridhwana for the mindful feedback on the expanded use of comments.  I have a lot of thoughts on this and want to approach this in a way where we're finding the sustainable path.

I see two topics in this comment:  The particular `@note` I wrote, which as you reflected started began to give "name" to the second topic; namely a concern for maintenance overhead paired with legibility of the codebase.

Setting the particular aside, I definitely confess that I often skip over comments and know that they require effort to keep synchronized with the code (e.g. the code is the canonical source of truth).  And sometimes I find comments a distraction, mostly in that I visually scan over them.  However, I think those "costs" are outweighed by the value they add and the tools we have to mitigate some of those costs.  I'm going to lightly touch on those and then look towards how we collectively can best approach.

- **Generated Docs**: When I run `yard; open doc/ApplicationPolicy.html` I see the API level documentation with the code collapsed.  This is a tool for our contributors to look at the application in a different way.  The conjecture being that the pairing of both documentation and code provides two different ways for folks to orient to the code.
- **Writing to Expand**: I use writing comments as my "final draft" of explaining the particulars of what I wrote (and my commit message as the synthesis of the particulars).  I have personally found in writing comments I sometimes nudge my method implementations to reflect a better understanding.  I _believe, but don't have research to support,_ that _well_ documented code has a higher reliability (which follows on the idea of revisiting the implementation to write the documentation).
- **Language Server Protocol (LSP) assistance**: With LSP, these comments now are available throughout the code; both during auto-completion and also when I hover on the class name or method name.  I don't know how VS Code exposes this, but LSP came about from Microsoft (creator and steward of VS Code) so I assume it's even better integrated.
- **Code Outline Tools**: I also use my code outline tools to provide a visual scan (I've always eschewed the mini map).  In VS Code I believe you can use "View > Open View... Outline" to add a pane that shows the file's outline (e.g. methods, constants, etc.)  This provides another way to scan without the distraction of method size nor comments.

Circling back to the particular example;  I think the `@note` should've been an `@todo`.  I like the idea of the "future improvement", with a caveat.  I see two concerns with the "future improvements".  One revolves around visibility and the other around proximity.

With forem being open source, we want to make sure we're sharing the future improvements that community members could tackle (e.g. improve this implementation detail).  The second is more delicate, and that is proximity.

With implementation details, the further something is from the implementing code, the greater the drift.  Method comments are already one degree removed.  If they are in a different file all together that's another degree.  And as they move outside the code-base, that greatly increases the drift.

These two concerns are solveable.

I put all of this forward to give you insight into my current thinking as well as to help facilitate a deliberate dialogue about understanding the aspirational guidelines of our in-line documentation.

Reply:
In discussion with @Ridhwana in another channel, we want to move the larger conversation to another place.  The particular of `@note` versus `@todo` versus omit is something to discuss on this PR.

Reply:
I'm of the opinion that if you've used the term "might" that this is not a concrete path outlined in the to do. Also based on "that's already a bit further afield than this need be."  seems like we're getting ahead of ourselves by adding this comment pre-emptively. Hence my vote is for _omit_ until/if we actually get to this point. 

Like I've mentioned earlier, this sounds like a refactor task that I would add to a project board so that we don't accumulate a ton of to-do's in our codebase that do not provide the necessary context. I think if a to-do is REALLY short lived it can possibly live in our codebase but otherwise it should be moved to a project board where we manage and track issues, otherwise the codebase just becomes unwieldy with to-do's.

Reading this out loud - "a future path might be to look at how controller actions do the 'respond_to do' blocks." I'm not actually concretely sure what needs to happen and why it needs to happen and also when it needs to happen. I'm also not sure how urgent it is, is it a short lived todo as part of a project or something for the future when we have time to investigate it. These questions could be answered in a project board task.

However, it's not a hill that I'm willing to die on 😅, we can deal with cases like this better if we use/create some best practices to our commenting in the future, and as we discussed, we'll continue that conversation separately.

Reply:
@Ridhwana I pushed up a modification to include the note regarding symmetry, but removed the prescribed todo and course of action.

I really dislike extracting the refactoring chore to a project board; but can see the utility if it's something that helps make tech debt visible for decision making.